### ISSUE TITLE
Cannot pickle 'SSLContext' object

### ISSUE THREAD

**Issue by itsamirhn** [Mahyar24/findsub#1]
I get this error on any try:
`cannot pickle 'SSLContext' object`

**Comment by Mahyar24** [comment]
Well, I need to see the full traceback and some platforms and versions info of your OS and Python interpreter. :)

**Comment by itsamirhn** [comment]
```
Traceback (most recent call last):
  File "/findsub/findsub/__main__.py", line 156, in run
    main(
  File "/findsub/findsub/__main__.py", line 83, in main
    process.start()
  File "/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: cannot pickle 'SSLContext' object
```
It seems about sth inside `extract_audio` function.

And I'm using MacOS 12.3.1 and Python 3.9.12

**Comment by Mahyar24** [comment]
Thank you, @itsamirhn, for the details. 
This is quite odd bug! My primary guess is that when the multiprocessing module wants to pickle the `extract_audio` function, it tries to pickle the `Movie` type too
https://github.com/Mahyar24/findsub/blob/e2a0dd682fdabb2bbe7c8a3612cc84581a0f8ed9/findsub/ffmpeg.py#L67
and when we look at the imports of the `movie` module
https://github.com/Mahyar24/findsub/blob/e2a0dd682fdabb2bbe7c8a3612cc84581a0f8ed9/findsub/movie.py#L18
we will face this external package. Which is almost certainly based on some HTTPS library for making IMDB requests that are not pickleable (IDK why!).
So I try to import the `Movie` type only when there is some type checking (which leads to the compilation of the `Movie` module and therefore `IMDB` package); otherwise, it will be just a generic type. (https://github.com/Mahyar24/findsub/commit/e2a0dd682fdabb2bbe7c8a3612cc84581a0f8ed9)


**Be aware that this commit doesn't come with a new release because `findsub` is not working anymore. Subscene is now under severe Cloudflare protection, and I tried to bypass it, which was successful at first. but besides getting a `200` response, there is also a rate limiter that eliminates the chance of downloading all subtitles with a reasonable speed (429 HTTP Error), so I have decided to deprecate this project for now.**
